### PR TITLE
Add user.create_saml_account and user.update_saml_account

### PR DESCRIPTION
**Pull request by reiterl**

### CODE REVIEW

**Comment by ManniLoe** [comment]
@reiterl Shouldn't we at least define the first and last name as a mandatory field?

File: openslides_backend/action/actions/user/create_saml_account.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    skip_archived_meeting_check = True
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        instance["username"] = instance["saml_id"]
+        if self.datastore.exists(
+            "user", FilterOperator("saml_id", "=", instance["saml_id"])
+        ):
+            raise ActionException("Saml_id already exists.")
+
+        filter_operators: List[FilterOperator] = []
```
Comment (r-peschke) [inline openslides_backend/action/actions/user/create_saml_account.py:45]: Remove the check of uniqueness for `first_name`, `last_name` and `email`.

File: tests/system/action/user/test_create_saml_account.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+from tests.system.action.base import BaseActionTestCase
+
+
+class UserCreateSamlAccount(BaseActionTestCase):
+    def test_create_saml_account_correct(self) -> None:
```
Comment (r-peschke) [inline tests/system/action/user/test_create_saml_account.py:5]: This test is not needed, if we have the second one

File: tests/system/action/user/test_create_saml_account.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    def test_create_saml_account_saml_id_exists(self) -> None:
+        self.set_models({"user/78": {"username": "111222333", "saml_id": "111222333"}})
+        response = self.request(
+            "user.create_saml_account",
+            {"saml_id": "111222333", "first_name": "Max", "last_name": "Mustermann"},
+        )
+        self.assert_status_code(response, 400)
+        assert "Saml_id already exists." in response.json["message"]
+
+    def test_create_saml_account_user_exists(self) -> None:
```
Comment (r-peschke) [inline tests/system/action/user/test_create_saml_account.py:62]: We don't need this test anymore.
But we need a test to check for duplicate `username`.

File: tests/system/action/user/test_update_saml_account.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
+from tests.system.action.base import BaseActionTestCase
+
+
+class UserUpdateSamlAccount(BaseActionTestCase):
+    def test_update_saml_account_correct(self) -> None:
+        self.set_models({"user/78": {"username": "111222333", "saml_id": "111222333"}})
```
Comment (r-peschke) [inline tests/system/action/user/test_update_saml_account.py:6]: Set a different `username` and check that it doesn't changed

**Review by r-peschke — CHANGES_REQUESTED** [review]
Sorry, but changes in action description led to some changes

**Comment by r-peschke** [comment]
> @reiterl Shouldn't we at least define the first and last name as a mandatory field?

Open Slides don't need this, therefore we don't want to make them required. But the customer could and should fill the fields.

File: openslides_backend/action/actions/user/create_saml_account.py (hunk 1 above)
Comment (reiterl) [reply openslides_backend/action/actions/user/create_saml_account.py:45]: Removed.

File: global/meta/models.yml [hunk 5]
```diff
@@ -200,6 +200,12 @@ user:
     type: string
     required: true
     restriction_mode: A
+  saml_id:
+    type: string
+    restriction_mode: A
+  single_sign_on_only:
```
[the remark this answers is not in the document]
Comment (reiterl) [reply global/meta/models.yml:206]: Removed.

File: openslides_backend/action/actions/user/create_saml_account.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        for field in ("first_name", "last_name", "email"):
+            if instance.get(field):
+                filter_operators.append(FilterOperator(field, "=", instance.get(field)))
+        if self.datastore.exists("user", And(*filter_operators)):
+            raise ActionException("User with name and email already exists.")
+
+        instance["single_sign_on_only"] = True
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/user/create_saml_account.py:52]: Removed.

File: openslides_backend/action/actions/user/create_saml_account.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+            "pronoun",
+            "is_active",
+            "is_physical_person",
+        ],
+        title="create saml account schema",
+    )
+    skip_archived_meeting_check = True
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        instance["username"] = instance["saml_id"]
```
[the remark this answers is not in the document]
Comment (reiterl) [reply openslides_backend/action/actions/user/create_saml_account.py:39]: I use UsernameMixin here.

File: tests/system/action/user/test_create_saml_account.py (hunk 2 above)
Comment (reiterl) [reply tests/system/action/user/test_create_saml_account.py:5]: Removed then.

File: tests/system/action/user/test_update_saml_account.py (hunk 4 above)
Comment (reiterl) [reply tests/system/action/user/test_update_saml_account.py:6]: Okay, done.

File: openslides_backend/action/actions/user/create_saml_account.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            "pronoun",
+            "is_active",
+            "is_physical_person",
+        ],
+        title="create saml account schema",
+    )
+    skip_archived_meeting_check = True
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        instance["username"] = next(
```
Comment (r-peschke) [inline openslides_backend/action/actions/user/create_saml_account.py:40]: For one instance an iterator is a little bit oversized I think, a `self.generate_usernames([instance["saml_id"]])[0]` as used in user/create.py would be sufficient

**Review by r-peschke — APPROVED** [review]
IMO you should use the simpler and in this case more efficient expression, see below, instead the `next(iter(`, but if you like, you could stay with this

File: openslides_backend/action/actions/user/create_saml_account.py (hunk 8 above)
Comment (reiterl) [reply openslides_backend/action/actions/user/create_saml_account.py]: Okay, I use `self.generate_usernames([instance["saml_id"]])[0]`.
